### PR TITLE
refactor: Improve flake

### PR DESCRIPTION
**Pull request by Sigmanificient**
I noticed some minor changes to improve the flake.nix file:
- inlining `pkgs` declaration within the `forAllSystems`
- replacing `runCommand` with `WriteShellScriptBin`
- using flake exposed systems list (4 -> 11 supported systems)

### CODE REVIEW

**Comment by juliusmarminge** [comment]
cc @tim-smart you were the one setting this up initially, I have no idea how it works xdd

File: flake.nix [hunk 1]
```diff
@@ -1,36 +1,28 @@
...
-            buildInputs = [
-              bun
-              corepackEnable
-              node
-            ];
-          };
+        default = pkgs.mkShell {
+          packages = with pkgs; [
+            bun
+            corepackEnable
```
Comment (tim-smart) [inline flake.nix:22]: This can be removed and replaced with the "corepack" package now.

**Review by tim-smart — CHANGES_REQUESTED** [review]

File: flake.nix [hunk 2]
```diff
@@ -1,36 +1,28 @@
...
+            node
```
Comment (tim-smart) [inline flake.nix:23]: We can inline the node package and remove the `let` block.

**Comment by Sigmanificient** [comment]
Now this is tiny!

**Review by tim-smart — APPROVED** [review]

**Review by juliusmarminge — APPROVED** [review]

**Comment by juliusmarminge** [comment]
Thanks for the review Tim :)

**Pull request merged**